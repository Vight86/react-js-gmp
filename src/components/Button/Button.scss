@use '../../scss/variables';

.button {
  font-family: inherit;
  font-size: 2em;
  color: variables.$color-fivefold;
  font-weight: 500;
  text-transform: uppercase;
  text-align: center;
  height: 2.9em;
  line-height: 2.9em;
  border-radius: .2em;
  transition: all .3s;
  cursor: pointer;
}

.primary {
  @extend .button;
  background: variables.$color-primary;
  border: .1em solid transparent;

  &:hover {
    border: .1em solid variables.$color-primary;
    color: variables.$color-primary;
    background: variables.$color-tertiary;
  }
}

.secondary {
  @extend .button;
  background: variables.$color-tertiary;
  border: 0.1em solid currentColor;
  color: variables.$color-primary;

  &:hover {
    border: 0.1em solid variables.$color-primary;
    background: variables.$color-primary;
    color: variables.$color-tertiary;
  }
}

.tertiary {
  @extend .button;
  background: transparentize(variables.$color-quaternary, 0.68);
  height: 2.3em;
  line-height: 2.3em;
  color: variables.$color-primary;

  &:hover {
    background: transparentize(variables.$color-quaternary, 0.2);;
  }
}

.normal {
  @extend .button;
  width: 9em;
}

.wide {
  @extend .button;
  width: 11.5em;
}

.search {
  background: url("./search.svg") no-repeat center;
  width: 2.9em;
  height: 2.9em;
  font-size: 1rem;
  transform: scale(1);
  transition: all 0.3s;

  &:hover {
    transform: scale(1.3);
  }
}

.close {
  position: absolute;
  top: 3em;
  right: 3em;
  width: 2em;
  height: 2em;
  font-size: 1rem;
  background: url("close.svg") no-repeat center;
  transition: all 0.3s;

  &:hover {
    cursor: pointer;
    transform: rotate(90deg);
  }
}


