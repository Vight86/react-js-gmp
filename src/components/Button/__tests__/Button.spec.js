import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import Button from '../Button';

const mockOnClick = jest.fn();

describe('Button', () => {
  const setup = () => render(
    <Button
      onClick={mockOnClick}
      theme="primary"
      isWide
    >
      Button
    </Button>,
  );

  test('render primary theme correctly', () => {
    const { container } = setup();
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('invokes onClic callback correctly', () => {
    const { container } = setup();
    fireEvent.click(container.querySelector('button'));
    expect(mockOnClick).toHaveBeenCalled();
  });
});
