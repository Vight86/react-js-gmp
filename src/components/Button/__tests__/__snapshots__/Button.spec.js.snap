// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render primary theme correctly 1`] = `"<button type=\\"button\\" class=\\"primary wide\\">Button</button>"`;
